telegram/_chatboost.py

```diff
@@ -0,0 +1,408 @@
+#!/usr/bin/env python
+#
+# A library that provides a Python interface to the Telegram Bot API
+# Copyright (C) 2015-2024
+# Leandro Toledo de Souza <devs@python-telegram-bot.org>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser Public License for more details.
+#
+# You should have received a copy of the GNU Lesser Public License
+# along with this program.  If not, see [http://www.gnu.org/licenses/].
+"""This module contains the classes that represent Telegram ChatBoosts."""
+
+from typing import TYPE_CHECKING, Dict, Final, Optional, Sequence, Tuple, Type
+
+from telegram._chat import Chat
+from telegram._telegramobject import TelegramObject
+from telegram._user import User
+from telegram._utils.argumentparsing import parse_sequence_arg
+from telegram._utils.datetime import extract_tzinfo_from_defaults, from_timestamp
+from telegram._utils.types import JSONDict
+
+if TYPE_CHECKING:
+    from telegram import Bot
+
+
+class ChatBoostSource(TelegramObject):
+    """
+    Base class for Telegram ChatBoostSource objects. It can be one of:
+
+    :class:`telegram.ChatBoostSourcePremium`
+    :class:`telegram.ChatBoostSourceGiftCode`
+    :class:`telegram.ChatBoostSourceGiveaway`
+
+    Objects of this class are comparable in terms of equality. Two objects of this class are
+    considered equal, if their :attr:`source` is equal.
+
+    .. versionadded:: NEXT.VERSION
+
+    Args:
+        source (:obj:`str`): The source of the chat boost. Can be one of:
+            :attr:`~telegram.ChatBoostSource.PREMIUM`, :attr:`~telegram.ChatBoostSource.GIFT_CODE`,
+            or :attr:`~telegram.ChatBoostSource.GIVEAWAY`.
+
+    Attributes:
+        source (:obj:`str`): The source of the chat boost. Can be one of:
+            :attr:`~telegram.ChatBoostSource.PREMIUM`, :attr:`~telegram.ChatBoostSource.GIFT_CODE`,
+            or :attr:`~telegram.ChatBoostSource.GIVEAWAY`.
+    """
+
+    __slots__ = ("source",)
+
+    PREMIUM: Final[str] = "premium"  # TODO: Make it a reference to constants
+    """:const:`telegram.constants.ChatBoostSources.PREMIUM`"""
+    GIFT_CODE: Final[str] = "gift_code"
+    """:const:`telegram.constants.ChatBoostSources.GIFT_CODE`"""
+    GIVEAWAY: Final[str] = "giveaway"
+    """:const:`telegram.constants.ChatBoostSources.GIVEAWAY`"""
+
+    def __init__(self, source: str, *, api_kwargs: Optional[JSONDict] = None):
+        super().__init__(api_kwargs=api_kwargs)
+
+        # Required by all subclasses:
+        self.source: str = source
```

Comment:
once the constants for the sources are implemented, please convert to enum here like done e.g. for `Chat.type` (and add corresponding test)